File: DSA-Python/2-pointers/sum_target.py
from typing import List

def pair_sum_sorted(nums: List[int], target: int) -> List[int]:
    # Write your code here
    left = 0
    right = len(nums) - 1
    while(left < right):
        sum_val = nums[left] + nums[right]
        if sum_val > target:
            right -= 1
        elif sum_val < target:
            left += 1
        else:
            return [left, right]
    return []

File: DSA-Python/2-pointers/2 Sum/SumTarget.cs
public class Solution {
    public bool twoSum(int[] nums, int target) {
        // Your code goes here
        // nums is already sorted
        int left = 0;
        int right = nums.Length - 1;
        while(left < right)
        {
            int sum = nums[left]+ nums[right];
            if(sum == target)
                return true;
            else if (sum < target) left++;
            else right--;
        }
        return false;
    }
}